stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/MediaManager.kt

```diff
@@ -426,6 +437,14 @@ class MicrophoneManager(
         ifAudioHandlerInitialized { it.selectDevice(device?.toAudioDevice()) }
         _selectedDevice.value = device
 
+        if (device !is StreamAudioDevice.Speakerphone && mediaManager.speaker.isEnabled.value == true) {
+            mediaManager.speaker._status.value = DeviceStatus.Disabled
+        }
+
+        if (device is StreamAudioDevice.Speakerphone) {
+            mediaManager.speaker._status.value = DeviceStatus.Enabled
+        }
+
         if (device !is StreamAudioDevice.BluetoothHeadset && device !is StreamAudioDevice.WiredHeadset) {
             selectedDeviceBeforeHeadset = device
```

Comment:
The wording device is wrong and doesn't make sense. When we are setting for something happened in past instead we should use lastSelectedDevice 

```kotlin
selectedDeviceBeforeHeadset = device
```

instead it would be more meaningful if we write 
```kotlin
selectedDeviceBeforeHeadset = lastSelectedDevice
```

Reply:
Maybe you can use `nonHeadsetFallbackDevice` instetad of `selectedDeviceBeforeHeadset`